Jenkinsfile

```diff
@@ -1091,6 +1099,67 @@ pipeline {
                         }
                     }
                 }
+                stage('win-2004') {
+                    when {
+                        beforeAgent true
+                        expression { params.windows2004 }
+                    }
+                    environment {
+                        DOCKER_BUILDKIT        = '0'
+                        DOCKER_DUT_DEBUG       = '1'
+                        SKIP_VALIDATION_TESTS  = '1'
+                        SOURCES_DRIVE          = 'd'
+                        SOURCES_SUBDIR         = 'gopath'
+                        TESTRUN_DRIVE          = 'd'
+                        TESTRUN_SUBDIR         = "CI"
+                        WINDOWS_BASE_IMAGE     = 'mcr.microsoft.com/windows/servercore'
+                        WINDOWS_BASE_IMAGE_TAG = '2004'
+                    }
+                    agent {
+                        node {
+                            customWorkspace 'd:\\gopath\\src\\github.com\\docker\\docker'
+                            label 'windows-2004'
+                        }
+                    }
+                    stages {
+                        stage("Print info") {
+                            steps {
+                                sh 'docker version'
+                                sh 'docker info'
+                            }
+                        }
+                        stage("Run tests") {
+                            steps {
+                                powershell '''
+                                $ErrorActionPreference = 'Stop'
+                                Invoke-WebRequest https://github.com/moby/docker-ci-zap/blob/master/docker-ci-zap.exe?raw=true -OutFile C:/Windows/System32/docker-ci-zap.exe
+                                ./hack/ci/windows.ps1
+                                exit $LastExitCode
+                                '''
+                            }
+                        }
+                    }
+                    post {
+                        always {
+                            junit testResults: 'bundles/junit-report-*.xml', allowEmptyResults: true
+                            catchError(buildResult: 'SUCCESS', stageResult: 'FAILURE', message: 'Failed to create bundles.zip') {
+                                powershell '''
+                                $bundleName="win-2004-integration"
+                                Write-Host -ForegroundColor Green "Creating ${bundleName}-bundles.zip"
+
+                                # archiveArtifacts does not support env-vars to , so save the artifacts in a fixed location
+                                Compress-Archive -Path "${env:TEMP}/CIDUT.out", "${env:TEMP}/CIDUT.err", "${env:TEMP}/junit-report-*.xml" -CompressionLevel Optimal -DestinationPath "${bundleName}-bundles.zip"
```

Comment:
`${env:TEMP}/junit-report-*.xml` looks to be correct way to look those files as junit-report-unit-tests.xml ended up to artifacts and it contains right result:
```xml
<testcase classname="github.com/docker/docker/pkg/archive" name="TestChangesDirsEmpty" time="0.120000">
	<failure message="Failed" type="">=== RUN   TestChangesDirsEmpty&#xA;--- FAIL: TestChangesDirsEmpty (0.12s)&#xA;    changes_test.go:261: Reported changes for identical dirs: [{\dirSymlink C}]&#xA;</failure>
</testcase>
<testcase classname="github.com/docker/docker/pkg/archive" name="TestChangesDirsMutated" time="0.100000">
	<failure message="Failed" type="">=== RUN   TestChangesDirsMutated&#xA;--- FAIL: TestChangesDirsMutated (0.10s)&#xA;    changes_test.go:391: unexpected change &#34;C \\dirSymlink&#34; &#34;\\dirnew&#34;&#xA;</failure>
</testcase>
```